modules/integration/tests-integration/tests-amqp/src/test/java/org/wso2/mb/integration/tests/amqp/functional/QueueUserAuthorizationTestCase.java

```diff
@@ -0,0 +1,562 @@
+     * User3 tries to publish and consume messages. User3 succeeds.
+     *
+     * @throws IOException
+     * @throws XPathExpressionException
+     * @throws AndesAdminServiceBrokerManagerAdminException
+     * @throws URISyntaxException
+     * @throws SAXException
+     * @throws XMLStreamException
+     * @throws UserAdminUserAdminException
+     * @throws LoginAuthenticationExceptionException
+     * @throws LogoutAuthenticationExceptionException
+     * @throws JMSException
+     * @throws AndesClientConfigurationException
+     * @throws AndesClientException
+     * @throws NamingException
+     */
+    @Test(groups = {"wso2.mb", "queue"})
+    public void performQueuePermissionDifferentRolesAssignedPermissions()
+            throws IOException, XPathExpressionException,
+                   AndesAdminServiceBrokerManagerAdminException, URISyntaxException, SAXException,
+                   XMLStreamException, UserAdminUserAdminException,
+                   LoginAuthenticationExceptionException, LogoutAuthenticationExceptionException,
+                   JMSException, AndesClientConfigurationException, AndesClientException,
+                   NamingException {
+        // "superAdmin" refers to the admin
+        this.createPublishAndSubscribeFromUser("superAdmin", "authQueue7");
+
+        // Adding publish subscribe permissions of 'authQueue7' to 'pub_sub_queue_role' role.
+        QueueRolePermission queueRolePermission = new QueueRolePermission();
+        queueRolePermission.setRoleName(PUB_SUB_QUEUE_ROLE);
+        queueRolePermission.setAllowedToConsume(true);
+        queueRolePermission.setAllowedToPublish(true);
+        this.updateQueueRoleConsumePublishPermission("authQueue7", queueRolePermission);
+        log.info("Consume/Publish permissions updated for " + PUB_SUB_QUEUE_ROLE);
+
+        this.createPublishAndSubscribeFromUser("authUser3", "authQueue7");
+    }
+
+    /**
+     * User1 is in Role1 where there are create queue permissions.
+     * User3 is in Role2 where there are no create queue permissions.
+     * Admin creates a queue and then publishes and consumes messages.
+     * Admin assigns publishing and consuming permissions to Role2.
+     * User1 tries to publish and consume messages. User1 fails.
+     *
+     * @throws IOException
+     * @throws XPathExpressionException
+     * @throws AndesAdminServiceBrokerManagerAdminException
+     * @throws URISyntaxException
+     * @throws SAXException
+     * @throws XMLStreamException
+     * @throws UserAdminUserAdminException
+     * @throws LoginAuthenticationExceptionException
+     * @throws LogoutAuthenticationExceptionException
+     * @throws JMSException
+     * @throws AndesClientConfigurationException
+     * @throws AndesClientException
+     * @throws NamingException
+     */
+    @Test(groups = {"wso2.mb", "queue"}, expectedExceptions = JMSException.class, expectedExceptionsMessageRegExp = ".*Permission denied.*")
```

Comment:
Fixed in https://github.com/hemikak/product-mb/commit/5cf92c05e10227615f2324a2a2b33201670f53c7